internal/server/recon/recognize.go

```diff
@@ -451,24 +461,86 @@ func getNumSpansForContainedIn(spans []*pb.TokenSpans_Span, startIdx int) int {
 	return 0
 }
 
-func combineContainedInSingle(
-	spans []*pb.TokenSpans_Span, startIdx, numSpans int) *pb.TokenSpans_Span {
+func combineTokenSpans(spans []*pb.TokenSpans_Span, startIdx, numSpans int) *pb.TokenSpans_Span {
 	startSpan := spans[startIdx]
-	endSpan := spans[startIdx+numSpans-1]
-
+	
 	res := &pb.TokenSpans_Span{Tokens: startSpan.Tokens}
 	for i := 1; i < numSpans; i++ {
 		res.Tokens = append(res.Tokens, spans[startIdx+i].GetTokens()...)
 	}
 
-	// This map is used to collect all the places for the combined span, with dedup.
+	return combineContainedInTokens(res, spans, startIdx, numSpans)
+}
+
+func getNextPlaceTokenSpan(spans []*pb.TokenSpans_Span, startIdx int) (*pb.TokenSpans_Span, int) {
+	for i := startIdx; i < len(spans); i++ {
+		if len(spans[i].GetPlaces()) > 0 {
+			return spans[i], i
+		}
+	}
+	return nil, -1
+}
+
+// Generates and returns two maps from the startSpan and nextSpan:
+// - StartSpan's DCID to the corresponding RecogPlace object.
```

Comment:
nit: mention that startspan can correspond to multiple places
nit: DCIDs of nextSpan's ...